### PR TITLE
Add new DFC vocabulary for order states

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

This pull request updates the used DFC context from 1.8.0 to 1.8.2 and adds a vocabulary including order states like Held and Complete.

#### What should we test?

- We could test the DFC Orders integration but maybe the specs are enough? @RaggedStaff ?

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: app/services/fdc_offer_broker.rb [hunk 1]
```diff
@@ -33,8 +33,7 @@ def wholesale_product(product_id)
     production_flow = catalog_item("#{product_id}/AsPlannedProductionFlow")
 
     if production_flow
-      wholesale_product_id = production_flow.product
-      catalog_item(wholesale_product_id)
+      production_flow.product
```
Comment (dacook) [inline app/services/fdc_offer_broker.rb:36]: I have to admit I don't understand why we needed to load the catalog item before, or why we don't need to now..

**Review by dacook — APPROVED** [review]
Looks good, apart from a small rubocop error.

**Review by dacook — CHANGES_REQUESTED** [review]
https://github.com/openfoodfoundation/openfoodnetwork/pull/12994/files?w=1#diff-178b1a2653d8faf263c1709797c21fbbb0bb8fc6ba0b02c17a2115ab47a483e6R54

File: app/services/fdc_offer_broker.rb (hunk 1 above)
Comment (mkllnk) [reply app/services/fdc_offer_broker.rb:36]: The `catalog_item` method looks up an object in the catalog, in this case it's returning the product. But with the updated context, the DFC importer knows that the URL is a product id and resolves that for us.

**Comment by mkllnk** [comment]
Rebased to get a spec fix in.

**Review by chahmedejaz — APPROVED** [review]
Looks good to me. Great work 👍

**Review by dacook — APPROVED** [review]

**Comment by dacook** [comment]
@mkllnk or @RaggedStaff is there any testing you'd like to do before merging?

**Comment by mkllnk** [comment]
> is there any testing you'd like to do before merging?

No. Merging.

**Pull request merged**